### ISSUE TITLE
Leaving known zone doesn't seem to put you into "not_home" state

### ISSUE THREAD

**Issue by cmsimike** [home-assistant/home-assistant-iOS#29]
Hi,

I have zone notifications on so I can debug when the app thinks I enter and leave a hass zone. This morning I noticed it said I left my home zone, and 6 minutes(ish) later it said I entered my work zone. Looking at the ui however doesn't indicate my device state changed to 'not_home' during that 6 minutes.

**Comment by cmsimike** [comment]
I noticed it put me into 'not_home' this morning so I guess it just isn't consistent?

**Comment by bagobones** [comment]
6 min commute to work? I wonder if it isn't just a case of background updates being very slow... Remember it only updates on significant location changes when in the background.

**Comment by cmsimike** [comment]
@bagobones i mean if the app can figure out if it left the zone, doesn't that imply the significant change happened? and it isn't a 6 minute commute - about 20 mins total, so it took some time for it to figure out i was out of my home zone but 6 mins after that it knew it was at work.

**Comment by bagobones** [comment]
Testing today I do see it update my location once on the way to work so I am not_home however the update is somewhere in-between work and home.. I have been at the office for over an hour and another update has not been sent apparently.


I am going to say this issue is that Background update AND GEOfencing are not working correctly.

**Comment by lucasweb78** [comment]
I'm seeing this issue as well. When I leave my home zone in the morning I get the `exited home` notification but do not see the location change pushed to hass for a considerable period. If I force a location change update then hass does update.

I do not see this issue when I enter my home zone, both the notification and hass update happen almost immediately and at the same time.

**Comment by gieljnssns** [comment]
I have the same problem...

**Comment by robbiet480** [comment]
Can everyone here please retest with 1.5.0 and report back? Thanks!

**Comment by ukgrant** [comment]
I was also experiencing this issue. After upgrading to 1.5.0 exiting zones doesn't seem to be firing at all.
Entering a zone works fine, but exiting a zone always produces "Not updating zone... because the database says nothing has changed" in the log.

App version 1.5.0
iOS 12
iPhone X
Update sources: Zone enter/exit only (all others disabled)

**Comment by ukgrant** [comment]
I installed 15.0 (17) this morning. Exiting a zone is now triggering. However "leaving home" automation's are not triggered.

When crossing the geo-fence boundary, the app sends a 'leave zone' event to home assistant with the coordinates of the centre of the zone. Home assistant shows the device as 'away', but inside the home region on the map, which is keeping the lights on.

Arguably this is an issue with Home Assistant and not the iOS app, where a "leave" event should be treated as "Unknown coordinates" rather than the center of the zone being left.

This is how Owntracks works around the problem [[link]](https://www.home-assistant.io/components/device_tracker.owntracks/):
- Location will be set to the center of the zone when you enter.
- Location will be set to GPS coordinates when you leave a zone.
- You may want to set your Zone radius in HA to be slightly smaller that the Owntracks region radius.

> Home Assistant will use the enter and leave messages to set your zone location. Your location will be set to the center of zone when you enter. Location updates from OwnTracks will be ignored while you are inside a zone.
> 
> When you exit a zone, Home Assistant will start using location updates to track you again. To make sure that Home Assistant correctly exits a zone (which it calculates based on your GPS coordinates), you may want to set your Zone radius in HA to be slightly smaller that the Owntracks region radius.

**Comment by ukgrant** [comment]
I have submitted a PR as I figured that changing the above automation to the 'state' platform works for 'Home' but not for 'Work'.

```
trigger:
    platform: state
    entity_id: device_tracker.grants_iphone
    from: 'home'
    to: 'not_home'
```

The next step is to also send the actual coordinates of the device so that we can use the 'event: leave' tigger with region monitoring.

**Comment by robbiet480** [comment]
@ukgrant reported this was fixed in #171 and then confirmed fixed by #178, closing as such!
